libpod/events.go

```diff
@@ -77,6 +77,11 @@ func (c *Container) newContainerEventWithInspectData(status events.Status, inspe
 		e.HealthStatus = containerHealthStatus
 	}
 
+	if status == events.Remove {
+		exitCode, _ := c.runtime.state.GetContainerExitCode(c.ID())
```

Comment:
@Luap99 what kind of check should be added here to check for error from `GetContainerExitCode`

Reply:
I think it is fine to ignore the error here but IMO we should not display the `ContainerExitCode` in the events when there is no exit code. However because `0` is a valid exit code we cannot use it with the current omitempty json logic.
While this is a pre-existing problem I think we should fix it. Having `-1` for no exit code might be fine but I don't particular like this either. IMO it would make much more sense to change  `ContainerExitCode` to `*int` then we have a proper `nil` value for no exit code which also means the omitempty json will correctly write it for `0` and only exclude it for `nil`. 

This ensures it will not show up the the events output and thus not change the behaviour because `"ContainerExitCode":-1` looks odd.